pkg/kubectl/cmd/edit.go

```diff
@@ -220,8 +221,7 @@ func RunEdit(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []strin
 				return fmt.Errorf("The edited file had a syntax error: %v", err)
 			}
 
-			// annotate the edited object for kubectl apply
-			if err := kubectl.UpdateApplyAnnotation(updates); err != nil {
+			if err := kubectl.CreateOrUpdateAnnotation(cmdutil.GetFlagBool(cmd, cmdutil.ApplyAnnotationsFlag), updates); err != nil {
```

Comment:
@jackgr after updating annotations of "updates" (`Info` of the edited file), the config annotation is added to it successfully. However, as you can see, only its data, "edited", will be used to generate a "patch" (see https://github.com/janetkuo/kubernetes/blob/apply-annotation-optional/pkg/kubectl/cmd/edit.go#L244). So the "patch" won't contain the config annotation we added earlier while we do Patch(), see https://github.com/janetkuo/kubernetes/blob/apply-annotation-optional/pkg/kubectl/cmd/edit.go#L264.